src/test/java/integ/basic/BasicProcess.java

```diff
@@ -0,0 +1,107 @@
+package integ.basic;
+
+import static integ.basic.BasicProcess.INPUT;
+import static integ.basic.BasicProcess.STATE_ID_NEXT_1;
+import static integ.basic.BasicProcess.STATE_ID_NEXT_2;
+
+import io.xdb.core.process.Process;
+import io.xdb.core.state.AsyncState;
+import io.xdb.core.state.AsyncStateOptions;
+import io.xdb.core.state.StateDecision;
+import io.xdb.core.state.StateMovement;
+import io.xdb.core.state.StateSchema;
+import io.xdb.gen.models.CommandRequest;
+import io.xdb.gen.models.CommandWaitingType;
+import lombok.NonNull;
+import org.junit.jupiter.api.Assertions;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BasicProcess implements Process {
+
+    public static final String STATE_ID_NEXT_1 = "STATE_ID_NEXT_1";
+    public static final String STATE_ID_NEXT_2 = "STATE_ID_NEXT_2";
+    public static final Integer INPUT = 11;
+
+    @Override
+    public @NonNull StateSchema getStateSchema() {
+        return StateSchema.withStartingState(new BasicStartingState(), new NextState1(), new NextState2());
+    }
+}
+
+class BasicStartingState implements AsyncState<Integer> {
+
+    @Override
+    public Class<Integer> getInputType() {
+        return Integer.class;
+    }
+
+    @Override
+    public CommandRequest waitUntil(final Integer input) {
+        System.out.println("BasicStartingState.waitUntil: " + input);
+        Assertions.assertEquals(INPUT, input);
+
+        return new CommandRequest().waitingType(CommandWaitingType.EMPTYCOMMAND);
```

Comment:
nit: it would be better if it provide a helper:
```
CommandRequest.empty();

or 

CommandRequest.empty;
```

Reply:
True, when we are ready for other CommandWaitingTypes, we should add helper like StateDecision. I will add it that time.

Reply:
Yeah no problem. We can always change it later before final launch. When it’s time to share with more people we will have another review to refine the APIs